sdk/src/instructions/composites/collect-all-txn.ts

```diff
@@ -138,11 +142,47 @@ export async function collectAllForPositionsTxns(
   const latestBlockhash = await ctx.connection.getLatestBlockhash();
   const txBuilders: TransactionBuilder[] = [];
 
-  let posIndex = 0;
+  // build tasks
+  // For TokenProgram-TokenProgram pair pool, collectFees and 3 collectReward instructions can be packed into one transaction.
+  // But if pool has TokenExtension, especially TransferHook, we can no longer pack all instructions into one transaction.
```

Comment:
The situation would improve if ALTs can be used.
It has been a long time since Versioned Transaction was released and I feel it is time for the UI (including supported wallet) and SDK to stop supporting legacy transactions.
Looking at an analysis of the wallet apps of Orca users, many users use Phantom.